src/System.Drawing.Common/src/System/Drawing/Internal/GPStream.cs

```diff
@@ -212,7 +216,7 @@ public unsafe void Write(byte* pv, uint cb, uint* pcbWritten)
 
             // Stream Span API isn't available in 2.0
 #if netcoreapp20
-            byte[] buffer = ArrayPool<byte>.Shared.Rent(4096);
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(Math.Max(DefaultBufferSize, checked((int)cb)));
```

Comment:
I don't understand the use of Max here.  If `cb` is smaller than DefaultBufferSize, since we're only going to work with at most `cb` bytes, why do we want to request a buffer larger than `cb`?  In other words, why aren't these just:
```C#
byte[] buffer = ArrayPool<byte>.Shared.Rent(checked((int)cb));
```
?

Reply:
That is true, plus we're only reading from the stream cb bytes. Will fix in a follow up PR.